### PR TITLE
feat: added coverage tests for all 10 chosen functions

### PR DESCRIPTION
**Pull request by oolofsson**
resolves issue #16

### CODE REVIEW

File: okio/src/main/java/okio/Base64.java [hunk 1]
```diff
@@ -43,31 +46,39 @@ private Base64() {
...
         //  z    122   51 (ASCII - 71)
         bits = c - 71;
       } else if (c >= '0' && c <= '9') {
+        Coverage.visitedge("Base64::decode", "if4");
         // char ASCII value
         //  0    48    52
         //  9    57    61 (ASCII + 4)
         bits = c + 4;
       } else if (c == '+' || c == '-') {
+        Coverage.visitedge("Base64::decode", "if4");
```
Comment (Abdallahassan) [inline okio/src/main/java/okio/Base64.java]: You have two "if4"
It's probably easier to change them to "if4a" and "if4b" so you don't have to change all the subsequent ones

File: okio/src/main/java/okio/Base64.java [hunk 2]
```diff
@@ -85,22 +97,28 @@ private Base64() {
...
       word = word << 6;
       out[outCount++] = (byte) (word >> 16);
       out[outCount++] = (byte) (word >> 8);
     }
 
     // If we sized our out array perfectly, we're done.
-    if (outCount == out.length) return out;
-
+    if (outCount == out.length){
+      Coverage.visitedge("Base64::decode", "if10");
```
Comment (Abdallahassan) [inline okio/src/main/java/okio/Base64.java]: two "if10"

File: okio/src/main/java/okio/Utf8.java [hunk 3]
```diff
@@ -79,50 +80,67 @@ public static long size(String string) {
...
     if (endIndex > string.length()) {
+      Coverage.visitedge("Utf8::size", "endindex>length");
       throw new IllegalArgumentException(
           "endIndex > string.length: " + endIndex + " > " + string.length());
     }
 
     long result = 0;
+
     for (int i = beginIndex; i < endIndex;) {
+      Coverage.visitedge("Utf8::size", "endindex<beginindex");
```
Comment (Abdallahassan) [inline okio/src/main/java/okio/Utf8.java]: There's another "endindex<beginindex" at line 93

File: okio/src/main/java/okio/Buffer.java [hunk 4]
```diff
@@ -1364,32 +1556,55 @@ Segment writableSegment(int minimumCapacity) {
...
+            loop2Done = true;
+          }
           s = s.next;
           offset = nextOffset;
         }
       }
     }
 
     // Scan through the segments, searching for b.
+    boolean loop3Done = false;
```
Comment (Abdallahassan) [inline okio/src/main/java/okio/Buffer.java:1591]: Can you briefly explain the purpose of the loop2Done thing?

**Review by Abdallahassan — CHANGES_REQUESTED** [review]
There were a few mistaked that I found, writing the same branchID for different branches in the same fuunction. Also, I realized that it's a pain in the ass to write all the branchIDs in the Coverage class when you initialize the data, I didn't even bother reviewing that part, so I'm probably going to change the class so that it automatically adds a new branchID whenever Coverage.visitedge(...) is called on a new branch.

Edit: or actually that won't work because branches that aren't covered will never be included. But maybe I should make it so that it throws an exception if you call Coverage.visitedge(...) with a branchID that doesn't exist, because then you would know that you forgot to include it in the initialization.

File: okio/src/main/java/okio/Base64.java (hunk 1 above)
Comment (oolofsson) [reply okio/src/main/java/okio/Base64.java]: Ok! ill fix this.

File: okio/src/main/java/okio/Buffer.java (hunk 4 above)
Comment (oolofsson) [reply okio/src/main/java/okio/Buffer.java:1591]: I used this for the previous version, when we were reading, replacing and writing to a file for every "visitedge". It is not needed now.

**Pull request merged**